modules/ppcp-wc-gateway/src/Gateway/class-processpaymenttrait.php

```diff
@@ -71,7 +72,15 @@ public function process_payment( $order_id ) {
 			}
 
 			$this->session_handler->destroy_session_data();
+		} catch ( RuntimeException $error ) {
+			$this->session_handler->destroy_session_data();
+			wc_add_notice(
+				__( 'Payment provider declined the payment, please use a different payment method.', 'woocommerce-paypal-payments' ),
+				'error'
+			);
+			return null;
```

Comment:
I would put the message as the exception message. This way, if we need to throw exceptions for other reasons, later on, we would not have to touch this piece of code anymore.

`throw new RuntimeException(__( 'Payment provider declined the payment, please use a different payment method.', 'woocommerce-paypal-payments' ));`

`wc_add_notice($error->getMessage())`

Reply:
Done, good point, thanks.